From cb81412bb621b3522e9da5ffb80297664c88f94b Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Mon, 15 Apr 2024 10:49:04 +0200
Subject: [PATCH 0457/1695] scpfw: module: voltage_domain: remove const
 attribute from config data

Removes the const attributes used for drvier_id and api_id fields of
struct mod_voltd_dev_config to allow building SCP-firmware voltage
domains configuration data during initialization instead of using
C source tables defined at build time.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 0a828d0940127bd2d8a2b550bfb5849602165cff scpfw: module: voltage_domain: remove const attribute from config data]
Change-Id: I7fb4cd455b55ffb13d42a44a7a0354be8eab5719
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 .../module/voltage_domain/include/mod_voltage_domain.h        | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/lib/scmi-server/SCP-firmware/module/voltage_domain/include/mod_voltage_domain.h b/core/lib/scmi-server/SCP-firmware/module/voltage_domain/include/mod_voltage_domain.h
index 85b096060..cd0c21f1f 100644
--- a/core/lib/scmi-server/SCP-firmware/module/voltage_domain/include/mod_voltage_domain.h
+++ b/core/lib/scmi-server/SCP-firmware/module/voltage_domain/include/mod_voltage_domain.h
@@ -78,10 +78,10 @@ enum mod_voltd_voltage_level_type {
  */
 struct mod_voltd_dev_config {
     /*! Reference to the device element within the associated driver module */
-    const fwk_id_t driver_id;
+    fwk_id_t driver_id;
 
     /*! Reference to the API provided by the device driver module */
-    const fwk_id_t api_id;
+    fwk_id_t api_id;
 };
 
 /*!
-- 
2.47.2

